tokio/src/net/tcp/into_split.rs

```diff
@@ -0,0 +1,221 @@
+//! `TcpStream` owned split support.
+//!
+//! A `TcpStream` can be split into an `IntoReadHalf` and a
+//! `IntoWriteHalf` with the `TcpStream::into_split` method.
+//! `IntoReadHalf` implements `AsyncRead` while `IntoWriteHalf`
+//! implements `AsyncWrite`.
+//!
+//! Compared to the generic split of `AsyncRead + AsyncWrite`, this specialized
+//! split has no associated overhead and enforces all invariants at the type
+//! level.
+
+use crate::future::poll_fn;
+use crate::io::{AsyncRead, AsyncWrite};
+use crate::net::TcpStream;
+
+use bytes::Buf;
+use std::error::Error;
+use std::mem::MaybeUninit;
+use std::net::Shutdown;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use std::{fmt, io};
+
+/// Owned read half of a [`TcpStream`], created by [`into_split`].
+///
+/// [`TcpStream`]: TcpStream
+/// [`into_split`]: TcpStream::into_split()
+#[derive(Debug)]
+pub struct IntoReadHalf(Arc<TcpStream>);
```

Comment:
Why can't they just be `ReadHalf`/`WriteHalf`? I know we used `Into`, but the `Into` prefix sounds more to me like some conversion is still ongoing.
Another option could be `OwnedReadHalf`. We could have made the normal `split` returning `BorrowedReadHalf`s, but I guess it's too late for that.

Reply:
I'm fine with using `OwnedReadHalf`.